compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt

```diff
@@ -57,12 +60,37 @@ class AdaptersProcessingStep(
         }
     }
 
+    @Suppress("UNCHECKED_CAST")
     private fun generateJsonAdapter(globalConfig: GlobalConfig, element: Element) {
         val nameAllocator = NameAllocator()
         val typeElement = MoreElements.asType(element)
         val typeMirror = typeElement.asType()
         val typeName = TypeName.get(typeMirror)
 
+        val metadata = typeElement.annotationMirrors
+                .first { MoreElements.asType(it.annotationType.asElement()).qualifiedName.contentEquals("kotlin.Metadata") }
+
+        val d1Key = metadata.elementValues.keys
+                .first { it.simpleName.contentEquals("d1") }
+        val d1 = metadata.elementValues[d1Key]!!.accept(KotshiAnnotationValueVisitor, null) as List<String>
+        val d2Key = metadata.elementValues.keys
+                .first { it.simpleName.contentEquals("d2") }
+        val d2 = metadata.elementValues[d2Key]!!.accept(KotshiAnnotationValueVisitor, null) as List<String>
+
+        val (nameResolver, classProto) = JvmProtoBufUtil.readClassDataFrom(d1.toTypedArray(), d2.toTypedArray())
+        val classMetadata: ClassMetadata = ClassMetadataImpl(classProto, nameResolver)
+
+        for ((name, constructor) in classMetadata.constructors) {
+            messager.printMessage(Diagnostic.Kind.WARNING, "name = $name", element)
```

Comment:
The name is the actual name of the parameter, this would allow us to not use `@Getter`